pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,464 @@
+                    }
+                    console.warn(jdata.message);
+                    hide_progress_message();
+                    $("#deploy-app-general-error-msg").text(err_msg).parent().show();
+                    
+                });
+        }
+
+        function create_everything(namespace, services, rcs, pods) {
+            var ns_json = {"apiVersion":"v1beta3","kind":"Namespace","metadata":{"name": "" }};
+            ns_json.metadata.name = namespace;
+            var tasks = [];
+
+            tasks.push([namespace, [JSON.stringify(ns_json)], client.create_ns]);
+
+            for (var serv in services) {
+                tasks.push([services[serv].metadata.name, [namespace, JSON.stringify(services[serv])], client.create_service]);
+            }
+
+            for (var rc in rcs) {
+                tasks.push([rcs[rc].metadata.name, [namespace, JSON.stringify(rcs[rc])], client.create_replicationcontroller]);
+            }
+
+            for (var p in pods) {
+                tasks.push([pods[p].metadata.name, [namespace, JSON.stringify(pods[p])], client.create_pod]);
+            }
+
+            var deferred = $.Deferred();
+            var total = tasks.length;
+
+            function step() {
+                var e = tasks.shift();
+                if (!e) {
+                    deferred.resolve();
+                    return;
+                }
+                var task_name = e[0];
+                var args = e[1];
+                var task = e[2];
+
+                task.apply(null, args).done(function(response) {
+                    deferred.notify("created", response, tasks.length - total, total);
+                    step();
+                }).fail(function(ex, response) {
+                    var jdata = JSON.parse(response);
+                    //skip for namespace
+                    if (ex.status == 409 && jdata.details.kind === "namespaces") {
+                        deferred.notify("skipped", response, tasks.length - total, total);
+                        step();
+                    } else {
+                        deferred.reject(ex, response);
+                    }
+                });
+            }
+
+            step();
+
+            return deferred.promise();
+        }
+
+        function show_progress_message() {
+            is_deploying.show();
+        }
+
+        function hide_progress_message() {
+            is_deploying.hide();
+        }
+
+        return {
+            'deploy': deploy
+        };
+
+    }
+
+    function deploy_app() {
+
+        var jsondata = "";
+        jsondata = appdeployer.jsondata;
+        var ns = get_ns_field_val();
+
+        if (jsondata === '') {
+            $('#deploy-app-manifest-file-empty').addClass('has-error').show();
+            valid_manifest = false;
+            display_deploy_button();
+            return;
+        }
+        if (ns.trim() === '' || ns.trim() === 'Custom Namespace' || !/^[a-z0-9]+$/i.test(ns.trim())) {
```

Comment:
Not sure what this is doing ... but "Custom Namespace" is translatable, so this will break.